File: python/43d_1325B_CopyCopyCopyCopyCopy.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        hset = set(input().split())
        ans.append(len(hset))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/43d_1325B_CopyCopyCopyCopyCopy.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, temp;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        std::unordered_set<int> hset;
        for(int j = 0; j < n; j++){
            std::cin>>temp;
            hset.insert(temp);
        }
        ans.push_back(hset.size());
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}
